### PR TITLE
[Feature][Connector-V2][Hbase] implement hbase catalog

### LINKED ISSUE #6780: [Feature][Hbase] Implement Hbase connector catalog and typeconverter
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Implement Hbase connector catalog and typeconverter
next work for https://github.com/apache/seatunnel/pull/6348

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhangshenghang**
### Purpose of this pull request

- resolve #6780 

### How was this patch tested?

Local And e2e

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java [hunk 1]
```diff
@@ -50,9 +53,13 @@
...
-        value = {},
-        type = {EngineType.SEATUNNEL},
-        disabledReason = "The hbase container authentication configuration is incorrect.")
+        value = {
+            TestContainerId.FLINK_1_13,
+            TestContainerId.FLINK_1_14,
+            TestContainerId.FLINK_1_15
+        },
+        type = {EngineType.SEATUNNEL, EngineType.SPARK},
+        disabledReason = "Test only one engine for first change")
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java]: Please remove it.

Comment (zhangshenghang) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java]: already remove

**Comment by Hisoka-X** [comment]
waiting #7169 be merged.

**Comment by Hisoka-X** [comment]
Please resolve conflicts.

File: seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/catalog/HbaseCatalog.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        checkNotNull(tablePath);
+        return hbaseClient.tableExists(tablePath.getTableName());
+    }
+
+    @Override
+    public CatalogTable getTable(TablePath tablePath)
+            throws CatalogException, TableNotExistException {
+        checkNotNull(tablePath, "tablePath cannot be null");
+        // Hbase cannot retrieve the columns of the table
+        return CatalogTable.of(
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/catalog/HbaseCatalog.java]: why not throw unsupportedexception? The catalogtable returned by this method is useless.

**Review by Hisoka-X — COMMENTED** [review]
Overall LGTM. I found a minior problem.

File: seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/catalog/HbaseCatalog.java (hunk 2 above)
Comment (zhangshenghang) [reply seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/catalog/HbaseCatalog.java]: Already modified

**Review by Hisoka-X — APPROVED** [review]
LGTM. Thanks @zhangshenghang !

**Review by hailin0 — APPROVED** [review]

**Pull request merged**